src/bin/wd_copier.rs

```diff
@@ -12,10 +15,67 @@ struct Opt {
     dst: PathBuf,
 }
 
+/// Get OS dirty byte count using procfs::Meminfo
+fn get_dirty_bytes() -> u64 {
+    match procfs::Meminfo::new() {
+        Ok(o) => o.dirty,
+        Err(_e) => 0,
+    }
+}
+
+/// Calculates the percentage of max as it approaches min
+///
+/// # Arguments
+///
+/// * `current` - current value of progress
+/// * `min` - target; percent = 100 when current <= min
+/// * `max` - starting point; percent = 0 when current == max
+fn calc_percent(current: u64, min: u64, max: u64) -> u64 {
+    // Subtract min from current but clamp to 0u64
+    let numerator = current.saturating_sub(min);
+    let denominator = max / 100;
+    100 - (numerator / denominator)
```

Comment:
potential division by zero here I think

Reply:
I think you're technically correct but the outer portion covers this scenario (due to the new error handling); https://github.com/Pobega/writedisk/blob/e5a078dcf9e903ab2f89c962fa3730357d1e57d2/src/bin/wd_copier.rs#L103

Reply:
Either way, is there an equivalent to `saturating_div` or a preferred way to handle this case? I don't think it's crazy to consider handling it explicitly even if the outer caller does.

Reply:
https://docs.rs/num/0.2.0/num/trait.CheckedDiv.html seems to be the most reasonable approach?

Reply:
Since you're dealing with integers here, I would think `if denominator == 0 { ... }` would be the most straightforward way to handle it. Although as I look at it, I'm not sure we should be dealing with ints here. I guess it probably works out since the numbers will be fairly high...

Reply:
The current solution I have locally is this:

```
-    let denominator = max / 100;
-    100 - (numerator / denominator)
+//    let denominator = max / 100;
+    let denominator = 0;
+    100 - match numerator.checked_div(denominator) {
+        Some(result) => result,
+        None => 50,
+    }
```

I could go with either solution since I'm pretty impartial to it, just let me know which way you prefer.

Reply:
8fb13e7